src/main/java/com/gmail/inverseconduit/bot/DefaultBot.java

```diff
@@ -62,14 +70,23 @@ private void processMessage(final ChatMessage chatMessage) {
     public Set<CommandHandle> getCommands() {
         return Collections.unmodifiableSet(commands);
     }
+    
+    public Set<CommandHandle> getListeners(){
+    	return Collections.unmodifiableSet(listeners);
+    }
 
     @Override
     public void subscribe(CommandHandle subscriber) {
-        commands.add(subscriber);
+    	if (subscriber.getName() == null){
+    		listeners.add(subscriber);
+    	} else {
+    		commands.add(subscriber);
+    	}
```

Comment:
Could be interesting, yes. This would be a step to extracting OakBot, though. Your call ;)